### PR TITLE
Personal/namalu/implement connection protocol

### PR DESCRIPTION
**Pull request by namalu**
Implemented Shell Auth Service and Soda Connection.

### CODE REVIEW

File: MHVLib/Authentication/MHVShellAuthService.m
Comment (miburf) [inline MHVLib/Authentication/MHVShellAuthService.m]: nit, currently

File: MHVLib/Authentication/MHVShellAuthService.m
Comment (miburf) [inline MHVLib/Authentication/MHVShellAuthService.m:188]: aibParamString?  this and mraBooleanString seem weird.  The param strings could be built as a dictionary, then a new [dictionary queryString] method that converts.

File: MHVLib/Authentication/MHVShellAuthService.m
Comment (miburf) [inline MHVLib/Authentication/MHVShellAuthService.m]: nit, instanceId

File: MHVLib/Configuration/MHVConfiguration.h
Comment (miburf) [inline MHVLib/Configuration/MHVConfiguration.h]: .NET :)

File: MHVLib/Configuration/MHVConfiguration.h
Comment (miburf) [inline MHVLib/Configuration/MHVConfiguration.h]: This doesn't sounds like it applies to app SDKs

File: MHVLib/Configuration/MHVConfiguration.h
Comment (miburf) [inline MHVLib/Configuration/MHVConfiguration.h]: This doesn't sounds like it applies to app SDKs

File: MHVLib/Configuration/MHVConfiguration.m
Comment (miburf) [inline MHVLib/Configuration/MHVConfiguration.m]: C# string formatter.

File: MHVLib/Connection/MHVSessionCredentialClient.m
Comment (miburf) [inline MHVLib/Connection/MHVSessionCredentialClient.m]: should now pass new "method" variable

File: MHVLib/Connection/MHVSodaConnection.m
Comment (miburf) [inline MHVLib/Connection/MHVSodaConnection.m]: should return; too.

File: MHVLib/Controllers/MHVBrowserAuthBroker.m
Comment (miburf) [inline MHVLib/Controllers/MHVBrowserAuthBroker.m]: nit, MHVBrowserAuthBroker

File: MHVLib/HVMobile/MHVServiceRequest.m
Comment (miburf) [inline MHVLib/HVMobile/MHVServiceRequest.m:106]: nit, {} aren't needed.  A few more below too.

**Review by miburf — APPROVED** [review]
A few things to fix before merging.

File: MHVLib/Authentication/MHVShellAuthService.m
Comment (namalu) [reply MHVLib/Authentication/MHVShellAuthService.m:188]: The initialism actually stands for Multi-Instance Aware. I'm going to leave this as is for now, until we have something working to unblock everyone else. We can come back and make improvements later.

**Pull request merged**
